### PR TITLE
Fix si.ibAllocated in FrozenObjectHeapManager

### PR DESCRIPTION
**Pull request by EgorBo**
`ibAllocated` is expected to be a real size of a segment, not a pointer to the end.

Regressed in https://github.com/dotnet/runtime/pull/90847

### CODE REVIEW

**Comment by EgorBo** [comment]
PTAL @cshung @tommcdon, thanks for help!

File: src/coreclr/vm/frozenobjectheap.cpp [hunk 1]
```diff
@@ -180,7 +180,7 @@ void FrozenObjectSegment::RegisterOrUpdate(uint8_t* current, size_t sizeCommited
         segment_info si;
         si.pvMem = m_pStart;
         si.ibFirstObject = sizeof(ObjHeader);
-        si.ibAllocated = (size_t)current;
+        si.ibAllocated = (size_t)current - (size_t)si.pvMem;
```
Comment (cshung) [inline src/coreclr/vm/frozenobjectheap.cpp:183]: You might want to `assert ((size_t)current >= (size_t)si.pvMem)`.

**Review by cshung — APPROVED** [review]
LGTM, thanks for the quick fix!

**Review by tommcdon — APPROVED** [review]
Thanks!

**Pull request merged**